Makefile

```diff
@@ -22,8 +22,8 @@ ifeq (vagrant, $(filter vagrant,$(USER) $(SUDO_USER)))
 	BIN_ARCH := -$(OS)
 endif
 REV :=$(shell git rev-parse --short=8 HEAD)
-TAG :=$(shell git for-each-ref --format="%(refname:short)" --sort=-authordate --count=1 refs/tags) # e.g. `v0.9.0`
-TAG_NUM :=$(shell git for-each-ref --format="%(refname:short)" --sort=-authordate --count=1 refs/tags | cut -c 2-) # e.g. `0.9.0`
+TAG :=$(shell git describe --tags --abbrev=0) # e.g. `v0.9.0`
+TAG_NUM :=$(shell git describe --tags --abbrev=0 | cut -c 2-) # e.g. `0.9.0`
```

Comment:
This includes the `-dev` suffix if present. Do you know if that's intentional from context?

Reply:
That said I don't find `REV`, `TAG`, or `TAG_NUM` used anywhere. I find duplicates or close approximations of the commands used here though.

Reply:
> This includes the `-dev` suffix if present. Do you know if that's intentional from context?

I believe it is intentional. Additionally, this change simply is preserving that behavior.

> That said I don't find `REV`, `TAG`, or `TAG_NUM` used anywhere. I find duplicates or close approximations of the commands used here though.

It looks like uses of these variables were removed in #6651, but the variables themselves were not cleaned up.

I'd like to cherry-pick this PR as-is, but will post a separate PR to remove the unused variables on `master`.